code/modules/atmospherics/machinery/bluespace_vendor.dm

```diff
@@ -0,0 +1,278 @@
+	var/tank_cost = 10
+	///Stores the current price of the gases inside the tank
+	var/gas_price = 0
+	///Helper for mappers, will automatically connect to the sender (ensure to only place one sender per map)
+	var/map_spawned = TRUE
+	///Base icon name for updating the appearance
+	var/base_icon = "bluespace_vendor"
+	///Current operating mode of the vendor
+	var/mode = BS_MODE_OFF
+
+//The one that the players make
+/obj/machinery/bluespace_vendor/built
+	map_spawned = FALSE
+	mode = BS_MODE_OPEN
+
+/obj/machinery/bluespace_vendor/north //Pixel offsets get overwritten on New()
+	dir = SOUTH
+	pixel_y = 30
+
+/obj/machinery/bluespace_vendor/south
+	dir = NORTH
+	pixel_y = -30
+
+/obj/machinery/bluespace_vendor/east
+	dir = WEST
+	pixel_x = 30
+
+/obj/machinery/bluespace_vendor/west
+	dir = EAST
+	pixel_x = -30
+
+/obj/machinery/bluespace_vendor/New(loc, ndir, nbuild)
+	. = ..()
+	if(ndir)
+		setDir(ndir)
+
+	if(nbuild)
+		panel_open = TRUE
+		pixel_x = (dir & 3)? 0 : (dir == 4 ? -30 : 30)
+		pixel_y = (dir & 3)? (dir == 1 ? -30 : 30) : 0
+
+	update_appearance()
+
+/obj/machinery/bluespace_vendor/Initialize()
+	. = ..()
+	AddComponent(/datum/component/payment, tank_cost, SSeconomy.get_dep_account(ACCOUNT_ENG), PAYMENT_ANGRY)
+
+/obj/machinery/bluespace_vendor/LateInitialize()
+	. = ..()
+	if(!map_spawned)
+		return
+	for(var/obj/machinery/atmospherics/components/unary/bluespace_sender/sender in GLOB.machines)
+		if(!sender)
+			continue
+		register_machine(sender)
+
+/obj/machinery/bluespace_vendor/Destroy()
+	unregister_machine()
+	return ..()
+
+/obj/machinery/bluespace_vendor/update_icon_state()
+	switch(mode)
+		if(BS_MODE_OFF)
+			icon_state = "[base_icon]_off"
+		if(BS_MODE_IDLE)
+			icon_state = "[base_icon]_idle"
+		if(BS_MODE_PUMPING)
+			icon_state = "[base_icon]_pumping"
+		if(BS_MODE_OPEN)
+			icon_state = "[base_icon]_open"
+	return ..()
+
+/obj/machinery/bluespace_vendor/process()
+	if(mode == BS_MODE_OPEN)
+		return
+	if(!selected_gas)
+		return
+	var/gas_path = gas_id2path(selected_gas)
+	connected_machine.bluespace_network.pump_gas_to(internal_tank.air_contents, (tank_filling_amount * 0.01) * 10 * ONE_ATMOSPHERE, gas_path)
+
+/obj/machinery/bluespace_vendor/multitool_act(mob/living/user, obj/item/multitool/multitool)
+	if(istype(multitool))
+		if(istype(multitool.buffer, /obj/machinery/atmospherics/components/unary/bluespace_sender))
+			if(connected_machine)
+				to_chat(user, "<span class='notice'>Changing [src] bluespace network...</span>")
+			if(!do_after(user, 0.2 SECONDS, src))
+				return
+			playsound(get_turf(user), 'sound/machines/click.ogg', 10, TRUE)
+			register_machine(multitool.buffer)
+			to_chat(user, "<span class='notice'>You link [src] to the console in [multitool]'s buffer.</span>")
+			return TRUE
+
+/obj/machinery/bluespace_vendor/attackby(obj/item/item, mob/living/user)
+	if(!pumping)
+		if(default_deconstruction_screwdriver(user, "[base_icon]_open", "[base_icon]_off", item))
+			check_mode()
+			return
```

Comment:
Similar to above avoid nesting and combine these when it doesn't hurt readability from too many conditions
```suggestion
	if(!pumping && default_deconstruction_screwdriver(user, "[base_icon]_open", "[base_icon]_off", item))
		check_mode()
		return
```